apps/frontend/src/app/3_organisms/ZeroLocForm/components/AdjustCreditLine.tsx

```diff
@@ -0,0 +1,421 @@
+    () =>
+      Math.max(
+        Number(existingCollateral) -
+          (newDebt *
+            (isRecoveryMode
+              ? CRITICAL_COLLATERAL_RATIO
+              : MINIMUM_COLLATERAL_RATIO)) /
+            rbtcPrice,
+        0,
+      ),
+    [existingCollateral, isRecoveryMode, newDebt, rbtcPrice],
+  );
+
+  const maxCollateralAmount = useMemo(
+    () =>
+      isIncreasingCollateral
+        ? maxCollateralToDepositAmount
+        : maxCollateralToWithdrawAmount,
+    [
+      isIncreasingCollateral,
+      maxCollateralToDepositAmount,
+      maxCollateralToWithdrawAmount,
+    ],
+  );
+
+  const maxBorrowAmount = useMemo(() => {
+    const collateral =
+      (collateralSize === 0 ? maxCollateralToDepositAmount : collateralSize) +
+      Number(existingCollateral);
+
+    const amount =
+      (collateral * rbtcPrice) /
+      (isRecoveryMode ? CRITICAL_COLLATERAL_RATIO : MINIMUM_COLLATERAL_RATIO);
+
+    const originationFee = getOriginationFeeAmount(amount, borrowingRate);
+
+    return Math.max(amount - originationFee - Number(existingDebt), 0);
+  }, [
+    borrowingRate,
+    collateralSize,
+    existingCollateral,
+    existingDebt,
+    isRecoveryMode,
+    maxCollateralToDepositAmount,
+    rbtcPrice,
+  ]);
+
+  const maxRepayAmount = useMemo(
+    () =>
+      Math.min(
+        Number(fromWei(_debtTokenWeiBalance)),
+        Number(existingDebt) - MIN_DEBT_SIZE,
+      ),
+    [_debtTokenWeiBalance, existingDebt],
+  );
+
+  const maxDebtAmount = useMemo(
+    () => (isIncreasingDebt ? maxBorrowAmount : maxRepayAmount),
+    [isIncreasingDebt, maxBorrowAmount, maxRepayAmount],
+  );
+
+  const originationFee = useMemo(() => {
+    if (isIncreasingDebt) {
+      return getOriginationFeeAmount(debtSize, borrowingRate);
+    }
+    return 0;
+  }, [borrowingRate, debtSize, isIncreasingDebt]);
+
+  const initialRatio = useMemo(
+    () =>
+      ((Number(existingCollateral) * rbtcPrice) / Number(existingDebt)) * 100,
+    [existingCollateral, existingDebt, rbtcPrice],
+  );
+
+  const ratio = useMemo(() => {
+    return ((newCollateral * rbtcPrice) / newDebt) * 100 || 0;
+  }, [newCollateral, newDebt, rbtcPrice]);
+
+  const { t } = useTranslation();
+
+  const initialLiquidationPrice = useMemo(
+    () =>
+      MINIMUM_COLLATERAL_RATIO *
+      (Number(existingDebt) / Number(existingCollateral)),
+    [existingDebt, existingCollateral],
+  );
+  const initialLiquidationPriceInRecoveryMode = useMemo(
+    () =>
+      CRITICAL_COLLATERAL_RATIO *
+      (Number(existingDebt) / Number(existingCollateral)),
+    [existingCollateral, existingDebt],
+  );
+
+  const liquidationPrice = useMemo(
+    () => MINIMUM_COLLATERAL_RATIO * (newDebt / newCollateral) || 0,
+    [newDebt, newCollateral],
+  );
+  const liquidationPriceInRecoveryMode = useMemo(
+    () => CRITICAL_COLLATERAL_RATIO * (newDebt / newCollateral) || 0,
+    [newDebt, newCollateral],
+  );
+
+  const errors = useMemo(() => {
```

Comment:
I think that this method could be extracted to a util function (doesn't need to be a custom hook) and it will have 2 benefits:
1. It will simplify the component - it's a large function that makes the component harder to read.
2. You won't need to duplicate code - it's almost identical to `errors` function in `OpenCreditLine.tsx`. The only difference is the first case and this can be adjusted.

Use the extracted function in both this file and `OpenCreditLine.tsx`.